File: flipflop.slome.org/2025/d03.py
DAY,_,_=__file__.rpartition('.')

from collections import Counter

r=[]

with open(f'{DAY}.txt','r')as F:
    for l in F:
        l=l.rstrip('\n')
        r+=[l]

print(Counter(r).most_common(1)[0][0])

s2=0
s3=0
for c in r:
    r,g,b=map(int,c.split(','))
    if r<g>b and r!=b:s2+=1

    if r==g or g==b or r==b:s3+=10
    elif g<r>b:s3+=5
    elif r<g>b:s3+=2
    elif g<b>r:s3+=4
print(s2)
print(s3)

File: CodingQuest/2024/d03.rb
DAY,_,_ = __FILE__.rpartition(?.)

require'prime'
require'z3'

r=[]
t=''
s=0

File.open("#{DAY}.txt", "r") { |f|
  f.each_line {|l|
    l=l.chomp
    #l=l.scan(/-?\d+/).map &:to_i
    #l=l.split.map &:to_i
    #l=l.chars.map &:to_i
    #l=l.split
    #l=l.to_i
    #l=l.chars
    
    t+=l
  }
}

l=''
i=0
t.split.map{l+='.#'[i^=1]*_1.to_i}
puts l.chars.each_slice(100).map{_1*''}